python/lsst/obs/base/butler_tests.py

```diff
@@ -0,0 +1,186 @@
+                        dimensions=None,
+                        sky_origin=None,
+                        raw_subsets=None,
+                        good_ccds=None,
+                        bad_ccds=None,
+                        linearizer=None
+                        ):
+        """
+        Set up the necessary variables for butlerGet tests.
+
+        ccdExposureId_bits: int
+            expected value of ccdExposureId_bits
+        exposureIds: dict
+            dict of exposure: exposure_id
+        filters: dict
+            dict of exposure: filter name
+        exptimes: dict
+            dict of exposure: exposure time
+        detector_names: dict
+            dict of exposure: detector name
+        detectorIds: dict
+            dict of exposure: detectorId
+        detector_serials: dict
+            dict of exposure: detector serial
+        dimensions: dict
+            dict of exposure: dimensions (as a geom.Extent2I)
+        sky_origin: tuple of float
+            Longitude, Latitude of 'raw' exposure
+        raw_subsets: tuple of (kwargs, int)
+            keyword args and expected number of subsets for butler.subset('raw', **kwargs)
+        good_ccds: list of int
+            list of valid ccd numbers
+        bad_ccds: list of int
+            list of invalid ccd numbers
+        linearizer: dict
+            'type':LinearizerType, and the converted values for 1,1000,30,000, for dataId['raw'],
+            or unittest.SkipTest to not test the linearizer.
+        """
+
+        fields = ['ccdExposureId_bits',
+                  'exposureIds',
+                  'filters',
+                  'exptimes',
+                  'detector_names',
+                  'detectorIds',
+                  'detector_serials',
+                  'dimensions',
+                  'sky_origin',
+                  'raw_subsets',
+                  'good_ccds',
+                  'bad_ccds',
+                  'linearizer'
+                  ]
+        ButlerGet = collections.namedtuple("ButlerGetData", fields)
+
+        self.butlerGetData = ButlerGet(ccdExposureId_bits=ccdExposureId_bits,
+                                       exposureIds=exposureIds,
+                                       filters=filters,
+                                       exptimes=exptimes,
+                                       detectorIds=detectorIds,
+                                       detector_names=detector_names,
+                                       detector_serials=detector_serials,
+                                       dimensions=dimensions,
+                                       sky_origin=sky_origin,
+                                       raw_subsets=raw_subsets,
+                                       good_ccds=good_ccds,
+                                       bad_ccds=bad_ccds,
+                                       linearizer=linearizer
+                                       )
+
+    def tearDown(self):
+        del self.butler
```

Comment:
The deletion should be done by the same class that takes hold of it.  Why not add a `ObsTests.setUp_tests` that will grab the butler+mapper, and destroy it in that?  Or use composition instead of multiple inheritance (because I think that's the root of your problems).